libs/service-portal/sessions/src/screens/Sessions/Sessions.tsx

```diff
@@ -1,22 +1,167 @@
-import { useAuth } from '@island.is/auth/react'
-import { Box, Text } from '@island.is/island-ui/core'
+import {
+  Box,
+  BreadCrumbItem,
+  Breadcrumbs,
+  GridColumn,
+  Hidden,
+  LoadingDots,
+  Text,
+  SkeletonLoader,
+  ToastContainer,
+  toast,
+  Input,
+} from '@island.is/island-ui/core'
 import { useLocale } from '@island.is/localization'
 import { IntroHeader } from '@island.is/portals/core'
+import React, { Fragment, useRef, useState } from 'react'
+import LogTable from '../../components/LogTable/LogTable'
+import LogTableMobile from '../../components/LogTable/LogTableMobile'
 
 import { m } from '../../lib/messages'
+import { useGetSessionsListQuery } from './Sessions.generated'
+import { SessionsSession } from '@island.is/api/schema'
+import { SessionsPaths } from '../../lib/paths'
+import * as kennitala from 'kennitala'
+import useOnScreen from '../../utils/useOnScreen'
+
+interface CursorState {
+  before: string
+  after: string
+}
 
 const Sessions = () => {
+  const SESSION_LIMIT = 20
+
   const { formatMessage } = useLocale()
-  const { userInfo } = useAuth()
+
+  const [sessionsData, setSessionsData] = useState<SessionsSession[]>([])
+  const [searchNationalId, setSearchNationalId] = useState('')
+  const [prevSearchNationalId, setPrevSearchNationalId] = useState('')
+  const [sentNationalId, setSentNationalId] = useState('')
+  const [page, setPage] = useState<CursorState>({ before: '', after: '' })
+  const [isSearchingSSN, setIsSearchingSSN] = useState(false)
+
+  const ref = useRef<HTMLDivElement>(null)
+  const isVisible = useOnScreen(ref)
+
+  const { data, loading, error } = useGetSessionsListQuery({
+    fetchPolicy: 'network-only',
+    variables: {
+      input: {
+        limit: SESSION_LIMIT,
+        before: page.before,
+        after: page.after,
+        nationalId: sentNationalId ?? '',
+        toDate: '',
+        fromDate: '',
+      },
+    },
+    onCompleted: (data) => {
+      setSessionsData([
+        ...(sessionsData as SessionsSession[]),
+        ...(data.sessionsList.data as SessionsSession[]),
+      ])
+      setIsSearchingSSN(false)
+    },
+    onError: () => {
+      toast.error(formatMessage(m.error))
+    },
+  })
```

Comment:
I started with using refetch but i quickly discovered that when it gets called it does not trigger changes on loading and data from the query hook itself which the process relies on. So that would mean I would need to keep track of those fields myself in state variables, or do you have any better suggestions how to handle that ?